dashboard/modules/reporter/reporter_agent.py

```diff
@@ -369,6 +372,25 @@ async def CpuProfiling(self, request, context):
         )
         return reporter_pb2.CpuProfilingReply(output=output, success=success)
 
+    async def MemoryProfiling(self, request, context):
+        pid = request.pid
+        format = request.format
+        leaks = request.leaks
+        duration = request.duration
+        native = request.native
+        p = MemoryProfilingManager(self._log_dir)
+        success, output = await p.attach_profiler(pid, native=native)
+        profiler_filename = output.split("\n")[-1]
+        if not success:
+            return reporter_pb2.MemoryProfilingReply(output=output, success=success)
+
+        await asyncio.sleep(duration + 1)  # add 1 second overhead
```

Comment:
what is this for?

Reply:
will add comment regarding memray possible overhead on this